src/views/Challenge.vue

```diff
@@ -79,17 +80,33 @@ export default {
         rating = 'rating'
         no_ass = 'no_assessments'
       }
-      return [
+      
+      const default_obj = [
         { k: { v: 'amount', r: false}, l: this.$t('challenge.REQUESTED_BUDGET_DESC')},
         { k: { v: 'amount', r: true}, l: this.$t('challenge.REQUESTED_BUDGET_ASC')},
         { k: { v: rating, r: false}, l: this.$t('challenge.REVIEW_SCORE_DESC')},
         { k: { v: rating, r: true}, l: this.$t('challenge.REVIEW_SCORE_ASC') },
         { k: { v: 'title', r: false}, l: this.$t('challenge.AZ') },
         { k: { v: 'title', r: true}, l: this.$t('challenge.ZA') },
         { k: { v: no_ass, r: false}, l: this.$t('challenge.NO_REVIEWS_DESC') },
-        { k: { v: no_ass, r: true}, l: this.$t('challenge.NO_REVIEWS_ASC') },
-        { k: { v: 'random', r: false}, l: this.$t('challenge.RANDOM') },
+        { k: { v: no_ass, r: true}, l: this.$t('challenge.NO_REVIEWS_ASC') }
       ]
+      let exists_vcast = proposals.filter((el) => el.votes_cast);
+      if (exists_vcast.length > 0){
```

Comment:
You can use the spread operator (`...`) to add elements to an iterable in combination with a condition, instead of using an external condition and pushing the new elements:

```
const default_obj = [
  { k: { v: 'amount', r: false}, l: this.$t('challenge.REQUESTED_BUDGET_DESC')},
  ...(this.withVoteCast ? [{ k: { v: 'votes_cast', r: false}, l: this.$t('challenge.NO_OF_WALLETS_DESC') }] : [])
]
```

It will keep code more compact and all the possible conditions in the same place.